pkg/operator/certrotation/target.go

```diff
@@ -102,6 +104,36 @@ func needNewTargetCertKeyPair(annotations map[string]string, caBundleCerts []*x5
 	return true
 }
 
+func needNewTargetCertKeyPairForTime(annotations map[string]string, signer *crypto.CA, validity time.Duration, renewalPercentage float32) bool {
+	targetExpiry := annotations[CertificateExpiryAnnotation]
+	if len(targetExpiry) == 0 {
+		return true
+	}
+	certExpiry, err := time.Parse(time.RFC3339, targetExpiry)
+	if err != nil {
+		glog.Infof("bad expiry: %q", targetExpiry)
+		// just create a new one
+		return true
+	}
+
+	// If Certificate is past its validity, we may must generate new.
+	if time.Now().After(certExpiry) {
+		return true
```

Comment:
shouldn't we start at 90% or so?

Reply:
> shouldn't we start at 90% or so?

caught below.  this is when it's straight out expired

Reply:
One could do a last try before 100%, even without the 10% check for the CA. but it's not clear it makes it much better.